src/Container.js

```diff
@@ -0,0 +1,431 @@
+    this.reset();
+  }
+  /**
+   * Handle the local errors
+   * @method onLocalError
+   * @private
+   * @param  {Event} $event Bellhop event
+   */
+  onLocalError($event) {
+    console.error('SpringRoll Container error: ', $event, new Error().stack);
+  }
+
+  /**
+   * Reset all the buttons back to their original setting
+   * and clear the iframe.
+   * @memberof Container
+   */
+  reset() {
+    const wasLoaded = this.loaded || this.loading;
+
+    // Destroy in the reverse priority order
+    if (wasLoaded) {
+      this.plugins
+        .slice()
+        .reverse()
+        .forEach(plugin => plugin.closed(this));
+    }
+
+    if (wasLoaded) {
+      CLIENT.trigger('closed');
+    }
+
+    // Reset state
+    this.loaded = false;
+    this.loading = false;
+
+    // Clear the iframe src location
+    this.main.setAttribute('src', '');
+    this.main.classList.remove('loading');
+  }
+
+  /**
+   * Set up communication layer between site and application.
+   * May be called from subclasses if they create/destroy Bellhop instances.
+   * @memberof Container
+   */
+  initClient() {
+    //Handle bellhop events coming from the application
+    CLIENT.on('loading', this.onLoading.bind(this));
+    CLIENT.on('loaded', this.onLoadDone.bind(this));
+    CLIENT.on('loadDone', this.onLoadDone.bind(this));
+    CLIENT.on('endGame', this.onEndGame.bind(this));
+    CLIENT.on('localError', this.onLocalError.bind(this));
+    // @ts-ignore
+    CLIENT.connect(this.dom);
+  }
+
+  /**
+   * If there was an error when closing, reset the container
+   * @memberof Container
+   */
+  _onCloseFailed() {
+    this.reset(); // force close the app
+  }
+
+  /**
+   * Open a application or path
+   * @param {string} userPath The full path to the application to load
+   * @param {Object} [userOptions] The open options
+   * @param {Boolean} [userOptions.singlePlay=false] If we should play in single play mode
+   * @param {Object | null} [userOptions.playOptions=null] The optional play options
+   * @memberof Container
+   */
+  _internalOpen(userPath, { singlePlay = false, playOptions = null } = {}) {
+    const options = { singlePlay, playOptions };
+    this.reset();
+
+    this.loading = true;
+    this.initClient();
+
+    const err = Features.basic();
+    if (err) {
+      console.error('ERROR:', err);
+      CLIENT.trigger('unsupported');
+    }
+    this.plugins.forEach(plugin => plugin.open(this));
+
+    let path = userPath;
+    if (null !== options.playOptions) {
+      const playOptionsQueryString =
+        'playOptions=' +
+        encodeURIComponent(JSON.stringify(options.playOptions));
+
+      path =
+        -1 === userPath.indexOf('?')
+          ? `${userPath}?${playOptionsQueryString}`
+          : `${userPath}&${playOptionsQueryString}`;
+    }
+
+    this.main.classList.add('loading');
+    this.main.setAttribute('src', path);
+
+    CLIENT.respond('singlePlay', { singlePlay });
+    CLIENT.respond('playOptions', { playOptions });
```

Comment:
This mismatches the [existing container behavior](https://github.com/SpringRoll/SpringRollContainer/blob/master/src/container/Container.js#L191).

This should instead be the plain object, and not the nested one:
```javascript
CLIENT.respond('singlePlay', singlePlay);
CLIENT.respond('playOptions', playOptions);
```

Reply:
We will need to update the Bellhop typing then, as it requires data to be a object